src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -50,7 +51,23 @@ public async Task<string> GetSecretValue(Uri secretUri, string key, Cancellation
             else if (client != null)
             {
                 KeyVaultSecret secret;
-                secret = await client.GetSecretAsync(secretName, secretVersion, cancellationToken).ConfigureAwait(false);
+                bool success = false;
+
+                try
+                {
+                    secret = await client.GetSecretAsync(secretName, secretVersion, cancellationToken).ConfigureAwait(false);
+                    success = true;
+                }
+                finally
+                {
+                    // If this is not already a cached secret, we don't have any refresh time to update,
+                    // i.e. there's no way to track next refresh time of new secrets that fail to resolve.
+                    if (!success && cachedSecret != null)
+                    {
+                        SetSecretInCache(key, secretValue: null, cachedSecret: cachedSecret, success: false);
+                    }
+                }
+
                 secretValue = secret?.Value;
                 SetSecretInCache(key, secretValue);
```

Comment:
If you agree that we should extend the expiration in all cases, this whole `else if` block can be simplified to 

```cs
try
{
    KeyVaultSecret secret = await client.GetSecretAsync(secretName, secretVersion, cancellationToken).ConfigureAwait(false);
    secretValue = secret?.Value;
}
finally
{
        SetSecretInCache(key, secretValue, cachedSecret);
}

```

Then in your `SetSecretInCache`, you create a new cached secret only if the passed in `cachedSecret` is null (the success case).